secboot/secboot_tpm.go

```diff
@@ -642,16 +727,21 @@ func efiImageFromBootFile(b *bootloader.BootFile) (sb_efi.Image, error) {
 // PCRHandleOfSealedKey retunrs the PCR handle which was used when sealing a
 // given key object.
 func PCRHandleOfSealedKey(p string) (uint32, error) {
-	r, err := sb_tpm2.NewFileSealedKeyObjectReader(p)
+	keyData, keyObject, err := readKeyFile(p)
 	if err != nil {
-		return 0, fmt.Errorf("cannot open key file: %v", err)
+		return 0, fmt.Errorf("cannot read key file %s: %w", p, err)
 	}
-	sko, err := sb_tpm2.ReadSealedKeyObject(r)
-	if err != nil {
-		return 0, fmt.Errorf("cannot read sealed key file: %v", err)
+	if keyObject != nil {
+		handle := uint32(keyObject.PCRPolicyCounterHandle())
+		return handle, nil
+	} else {
+		sealedKeyData, err := sb_tpm2.NewSealedKeyData(keyData)
+		if err != nil {
+			return 0, fmt.Errorf("cannot read key data in keyfile %s: %w", p, err)
+		}
+		handle := uint32(sealedKeyData.PCRPolicyCounterHandle())
+		return handle, nil
 	}
```

Comment:
tests don't seem to reach the if

Reply:
This one is a bit annoying. I would need to mock `PCRPolicyCounterHandle`, because there is no easy way to generate sealed objects from secboot public method. This code is very likely to change soon. Since we will be supposed to use the same handle all the time. The code was already not well covered because of the same reason. See the TODO in `TestPCRHandleOfSealedKey`. Is that OK if we do not cover it?

Reply:
I see, it's ok if it's not a regression in coverage